lib/ethereum-sync-protocol/src/lib.rs

```diff
@@ -386,441 +386,310 @@ pub fn next_sync_committee_gindex_at_slot<C: ChainSpec>(chain_id: u64, slot: Slo
+    }
+
+    pub const GENESIS_VALIDATORS_ROOT_6553725: H256 = H256::new(hex!(
+        "d8ea171f3c94aea21ebc42a1ed61052acf3f9209c00e4efbaaddac09ed9b8078"
+    ));
+
+    fn update_6553725(
+        update: &ethereum_sync_protocol_types::LightClientUpdate,
+    ) -> Result<(), Error> {
+        validate_light_client_update::<Mainnet, BlsVerifier>(
+            SEPOLIA_CHAIN_ID,
+            update,
+            Some(&SYNC_COMMITTEE_6553725),
+            None,
+            Slot::new(6553726),
+            Slot::new(6553718),
+            GENESIS_VALIDATORS_ROOT_6553725,
+            BlsVerifier,
+        )
+    }
+
+    #[test]
+    fn validate_update_works() {
+        assert_eq!(update_6553725(&UPDATE_6553725), Ok(()));
+    }
+
+    #[test]
+    fn validate_update_fails_when_sync_committee_is_wrong() {
+        let mut update = UPDATE_6553725.clone();
+        update.sync_aggregate.sync_committee_bits = Vec::new();
+
+        assert!(matches!(
+            update_6553725(&update),
+            Err(Error::InsufficientSyncCommitteeParticipants(..))
+        ));
+    }
+
+    #[test]
+    fn validate_update_fails_when_altered_next_sync_committee() {
+        let mut update = UPDATE_6553725.clone();
+        update.finalized_header.beacon.slot = update.finalized_header.beacon.slot + Slot::new(1);
+
+        let mut next_sync_committee = update.next_sync_committee.as_ref().unwrap().clone();
+        next_sync_committee.aggregate_pubkey = Default::default();
+
+        assert!(matches!(
+            validate_light_client_update::<Mainnet, BlsVerifier>(
+                SEPOLIA_CHAIN_ID,
+                &update,
+                Some(&SYNC_COMMITTEE_6553725),
+                Some(&next_sync_committee),
+                Slot::new(6553726),
+                Slot::new(6553718),
+                GENESIS_VALIDATORS_ROOT_6553725,
+                BlsVerifier,
+            ),
+            Err(Error::InvalidMerkleBranch(..))
+        ));
+    }
+
+    #[test]
+    fn validate_update_fails_when_altered_finalized_header() {
+        let mut update = UPDATE_6553725.clone();
+        update.finalized_header.beacon.slot = update.finalized_header.beacon.slot + Slot::new(1);
+
+        assert!(matches!(
+            update_6553725(&update),
+            Err(Error::InvalidMerkleBranch(..))
+        ));
+    }
+
+    /// This test also ensures that any `attested_header.beacon` alteration would cause the signature verification
+    /// to fail.
+    #[test]
+    fn validate_update_signature_data_correct() {
+        pub struct BlsVerifier;
+
+        impl BlsVerify for BlsVerifier {
+            const INVERSE: bool = false;
+
+            fn aggregate_verify_signature<'pk>(
+                &self,
+                aggregate_public_key: &'pk H384,
+                public_keys: impl IntoIterator<Item = &'pk H384>,
+                msg: Vec<u8>,
+                signature: H768,
+            ) -> Result<(), Error> {
+                let fork_version_slot =
```

Comment:
why is all of this logic done inside `aggregate_verify_signature`?

Reply:
this test doesn't do signature verification since it isn't the library's responsibility to contain the implementation of the signature verification. This checks whether the provided data to signature verification is valid for a certain scenario.